src/__tests__/scheduler.test.js

```diff
@@ -0,0 +1,56 @@
+require("../scheduler.js");
+
+describe("SubtitleScheduler", () => {
+    let scheduler;
+
+    beforeEach(() => {
+        debugger
+        scheduler = new sabre.SubtitleScheduler();
+    });
+
+    describe("#_eventListComparator", () => {
+        let SSASubtitleEventA;
+        let SSASubtitleEventB;
+
+        const eventAStartValue = 10;
+        const eventBStartValue = 6;
+        const eventsStartValueDiff = eventAStartValue - eventBStartValue;
+
+        const eventAEndValue = 5;
+        const eventBEndValue = 3;
+        const eventsEndValueDiff = eventAEndValue - eventBEndValue;
+
+        beforeEach(() => {
+            SSASubtitleEventA ={
+                getStart: jest.fn().mockName('getStart').mockReturnValue(eventAStartValue),
+                getEnd: jest.fn().mockName('getEnd').mockReturnValue(eventAEndValue),
+            };
+            SSASubtitleEventB ={
+                getStart: jest.fn().mockName('getStart').mockReturnValue(eventBStartValue),
+                getEnd: jest.fn().mockName('getEnd').mockReturnValue(eventBEndValue)
+            };
+        });
+
+        it("should return diff value between two events based on their start values if they are not the same", () => {
+            expect(scheduler._eventListComparator(SSASubtitleEventA, SSASubtitleEventB)).toBe(eventsStartValueDiff);
+
+            expect(SSASubtitleEventA.getStart).toBeCalledTimes(2);
```

Comment:
during the test I found that the production code can be optimized to reduce the count of `getStart` methods calls:
current implementation for `_eventListComparator`:
```
function (/** SSASubtitleEvent */ a, /** SSASubtitleEvent */ b) {
  if (a.getStart() === b.getStart()) return a.getEnd() - b.getEnd();
  return a.getStart() - b.getStart();
}
```

proposal:
```
function (/** SSASubtitleEvent */ a, /** SSASubtitleEvent */ b) {
  const eventAStartValue = a.getStart();
  const eventBStartValue = b.getStart();

  if (eventAStartValue  === eventAStartValue) {
    return a.getEnd() - b.getEnd();
  }

  return a.getStart() - b.getStart();
}
```

Since the method does not use instance related data (do not use **this**) it could be static method or arrow function if supports in used standard.